### PR TITLE
DOC: v6.0.0 release notes

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

I'm asking for reviews from all contributors as an opportunity to check that your work is properly represented here. There is no obligation to review.

This will be a new major version (v6.0.0). due to the API change. This PR combines the release notes from all the pre-release note pages.

## Motivation and Context

The start of the month is a good time to tag.

### CODE REVIEW

File: docs/source/releases.rst [hunk 1]
```diff
@@ -1,6 +1,76 @@
...
+v6.0.0 (2022-05-03)
+===================
+
+API Changes
+-----------
+- ``MultiDerivedSignal`` and ``MultiDerivedSignalRO`` calculation functions
+  (``calculate_on_get`` and ``calculate_on_put``) now take new signatures.
+  Calculation functions may be either methods on an ``ophyd.Device`` (with
+  ``self``) or standalone functions with the following signature:
+  .. code::
```
Comment (klauer) [inline docs/source/releases.rst:14]: I checked that this rendered properly when submitting that PR. I expect that it should be just fine here, too.

**Review by klauer — APPROVED** [review]
LGTM

**Comment by ZLLentz** [comment]
I'm actually going to merge this when the tests pass regardless of review- any review can easily result in amended release notes later.

**Review by tangkong — APPROVED** [review]
LGTM, lots of cool changes here.

**Pull request merged**